api/views.py

```diff
@@ -65,7 +66,42 @@ class RegistrationAPIViewSet(mixins.CreateModelMixin, viewsets.GenericViewSet):
     serializer_class = RegistrationSerializer
 
 
+class VolunteerFilter(filters.FilterSet):
+    time_volunteered = filters.NumberFilter(lookup_expr='exact')
+    time_volunteered__gt = filters.NumberFilter(method='time_volunteered_gt')
+    time_volunteered__lt = filters.NumberFilter(method='time_volunteered_lt')
+
+    def time_volunteered_gt(self, queryset, field_name, value):
+        return queryset.filter(time_volunteered__gt=value)
+
+    def time_volunteered_lt(self, queryset, field_name, value):
+        return queryset.filter(time_volunteered__lt=value)
+
+    class Meta:
+        model = Volunteer
+        fields = {
+            'id': ['exact'],
+            'first_name': ['exact', 'icontains'],
+            'last_name': ['exact', 'icontains'],
+            'tz_number': ['exact', 'icontains'],
+            'phone_number': ['exact', 'icontains'],
+            'date_of_birth': ['gt', 'lt', 'exact'],
+            'age': ['gt', 'lt', 'exact'],
+            'gender': ['exact'],
+            'city': ['exact', 'in'],
+            'neighborhood': ['exact', 'icontains'],
+            'areas': ['exact'],
+            'moving_way': ['exact'],
+            'week_assignments_capacity': ['exact', 'range'],
+            'wanted_assignments': ['exact'],
+            'score': ['exact'],
+            'created_date': ['gt', 'lt', 'exact'],
+            'organization': ['exact', 'in']
+        }
+
+
 class ListVolunteersViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
-    queryset = Volunteer.objects.all().order_by('-created_date')
+    queryset = Volunteer.objects.all().order_by('-created_date').annotate(time_volunteered=Count('helprequest'))
     serializer_class = VolunteerSerializer
-    permission_classes = [IsAuthenticated]
+    permission_classes = [AllowAny]
```

Comment:
Why is this set as AllowAny?

Reply:
Forgot to change it back. Used it for testing. My bad